ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -806,6 +807,523 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
+      }
+    }
+
+  private[this] val invisibleTransactionTreeByEventId =
+    LedgerTest(
+      "TXInvisibleTransactionTreeByEventId",
+      "Do not expose an invisible transaction tree by event identifier") { context =>
+      for {
+        Vector(alpha, beta) <- context.participants(2)
+        party <- alpha.allocateParty()
+        intruder <- beta.allocateParty()
+        dummy <- alpha.create(party, Dummy(party))
+        tree <- alpha.exercise(party, dummy.exerciseDummyChoice1)
+        _ <- synchronize(alpha, beta)
+        failure <- beta.transactionTreeByEventId(tree.rootEventIds.head, intruder).failed
+      } yield {
+        assertGrpcError(failure, Status.Code.NOT_FOUND, "Transaction not found, or not visible.")
+      }
+    }
+
+  private[this] val transactionTreeByEventIdInvalid =
+    LedgerTest(
+      "TXTransactionTreeByEventIdInvalid",
+      "Return INVALID when looking up an invalid transaction tree by event identifier") { context =>
+      for {
+        ledger <- context.participant()
+        party <- ledger.allocateParty()
+        failure <- ledger.transactionTreeByEventId("dont' worry, be happy", party).failed
+      } yield {
+        assertGrpcError(failure, Status.Code.INVALID_ARGUMENT, "Invalid field event_id")
+      }
+    }
+
+  private[this] val transactionTreeByEventIdNotFound =
+    LedgerTest(
+      "TXTransactionTreeByEventIdNotFound",
+      "Return NOT_FOUND when looking up an inexistent transaction tree by event identifier") {
+      context =>
+        for {
+          ledger <- context.participant()
+          party <- ledger.allocateParty()
+          failure <- ledger.transactionTreeByEventId(s"#${"a" * 60}:000", party).failed
+        } yield {
+          assertGrpcError(failure, Status.Code.NOT_FOUND, "Transaction not found, or not visible.")
+        }
+    }
+
+  private[this] val transactionTreeByEventIdNoParty =
+    LedgerTest(
+      "TXTransactionTreeByEventIdNotFound",
+      "Return INVALID_ARGUMENT when looking up a transaction tree by event identifier without specifying a party") {
+      context =>
+        for {
+          ledger <- context.participant()
+          failure <- ledger.transactionTreeByEventId("not-relevant").failed
+        } yield {
+          assertGrpcError(
+            failure,
+            Status.Code.INVALID_ARGUMENT,
+            "Missing field: requesting_parties")
+        }
+    }
+
+  private[this] val flatTransactionByEventId =
+    LedgerTest(
+      "TXFlatTransactionByEventId",
+      "Expose a visible flat transaction by event identifier") { context =>
+      for {
+        ledger <- context.participant()
+        party <- ledger.allocateParty()
+        dummy <- ledger.create(party, Dummy(party))
+        tree <- ledger.exerciseForFlatTransaction(party, dummy.exerciseDummyChoice1)
+        event = tree.events.head.event
+        eventId = event.created.map(_.eventId).getOrElse(event.archived.map(_.eventId).get)
```

Comment:
Hm, shouldn't we know precisely which event we expect? Although I guess in this case it doesn't matter so much, as we simply want to look up the transaction by any eventId.

Reply:
I believe the intent here was to not encode the order in which a transaction arrived in the test. I'm not sure this makes sense though and I would be in favor of pinning the expected transaction to be either a create or an archival. Makes sense?

Reply:
Right. The Ledger API doesn't specify the order or the events, but there should only be a single archived event in this case anyway.

Reply:
Of course, thanks.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2961/commits/962544588c5b708f7b2bd392347a46652d088d34